src/main/java/codeu/controller/ActivityFeedServlet.java

```diff
@@ -50,25 +53,80 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
     List<Activity> activities = activityStore.getActivities();
 
+    /*Sorts activities by creationTime field*/
+    Collections.sort(activities, new Comparator<Activity>() {
+      public int compare(Activity one, Activity other) {
+        return other.getCreationTime().compareTo(one.getCreationTime());
+      }
+    });
+
     request.setAttribute("activities", activities);
+    request.setAttribute("checked", "recent");
 
     request.getRequestDispatcher("/WEB-INF/view/activityfeed.jsp").forward(request, response);
 
-
   }
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
 
-    String username = request.getParameter("searchQuery");
+    List<Activity> activities = (List<Activity>) request.getSession().getAttribute("activities");
+    String toggle = "Undefined";
+    String sortingStyle = request.getParameter("sortingStyle") == null? "Undefined":request.getParameter("sortingStyle");
+    /*If the user posts a search request that isn't an empty string, reassembles the list of activities */
+    if (request.getParameter("searchQuery") != null && !request.getParameter("searchQuery").equals("")) {
+      String username = request.getParameter("searchQuery");
+
+      String cleanedUsername = Jsoup.clean(username, Whitelist.none());
+
+      activities = activityStore.getUserActivities(cleanedUsername);
+      request.setAttribute("searchQuery", cleanedUsername);
+
+      toggle = request.getParameter("checked");
+
+      if (toggle.equals("popular")) {
+        request.setAttribute("checked", "popular");
+      } else if (toggle.equals("recent")) {
+        request.setAttribute("checked", "recent");
+      }
+
+    }
+
+    /* If one of the sorting radio buttons are been toggled, sorts the list of activities*/
+
+    //if (request.getParameter("sortingStyle") != null) {
+
 
-    String cleanedUsername = Jsoup.clean(username, Whitelist.none());
 
-    List<Activity> activities = activityStore.getUserActivities(cleanedUsername);
+      if(sortingStyle.equals("popular") || toggle.equals("popular")) {
+
+        /*Sorts activities by allTimeCount field*/
+        Collections.sort(activities, new Comparator<Activity>() {
+          public int compare(Activity one, Activity other) {
+            return other.getAllTimeCount() - one.getAllTimeCount();
+          }
+        });
+
+        request.setAttribute("checked", "popular");
+
+      } else if (sortingStyle.equals("recent") || toggle.equals("recent")) {
+
+        /*Sorts activities by creationTime field*/
+        Collections.sort(activities, new Comparator<Activity>() {
```

Comment:
OK for now, but looks like you're duplicating the same code here as in the doGet() method, around line 57. I would probably make this a static inner class like "ActivityComparatorByCreationTime" so as to have it only once, and also make a "ActivityComparatorByAllTimeCount" for consistency. In fact I'd probably put the Comparators as inner classes in the Activity class itself.
All that said, while you are doing this correctly, keep in mind that anonymous inner classes like this are much less widely used nowadays. The "professional" way of doing this now would be to use lambdas. See https://docs.oracle.com/javase/tutorial/java/javaOO/lambdaexpressions.html if you aren't familiar. That's optional, but you'll make a better impression in an interview if you demonstrate that you know how to use them.